src/main/java/com/oltpbenchmark/benchmarks/templated/TemplatedBenchmark.java

```diff
@@ -0,0 +1,287 @@
+ *
+ */
+package com.oltpbenchmark.benchmarks.templated;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+import org.apache.commons.text.StringEscapeUtils;
+import org.codehaus.commons.compiler.CompilerFactoryFactory;
+import org.codehaus.commons.compiler.ICompilerFactory;
+import org.codehaus.commons.compiler.ISimpleCompiler;
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.oltpbenchmark.WorkloadConfiguration;
+import com.oltpbenchmark.api.BenchmarkModule;
+import com.oltpbenchmark.api.Loader;
+import com.oltpbenchmark.api.Procedure;
+import com.oltpbenchmark.api.SQLStmt;
+import com.oltpbenchmark.api.TransactionType;
+import com.oltpbenchmark.api.Worker;
+import com.oltpbenchmark.benchmarks.templated.procedures.GenericQuery;
+import com.oltpbenchmark.benchmarks.templated.procedures.GenericQuery.QueryTemplateInfo;
+import com.oltpbenchmark.benchmarks.templated.util.GenericQueryOperation;
+import com.oltpbenchmark.benchmarks.templated.util.TraceTransactionGenerator;
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+
+/**
+ * This class is used to execute templated benchmarks, i.e., benchmarks that
+ * have parameters that the user wants to set dynamically. A templated benchmark
+ * has the structure
+ * <query_templates>
+ *  <query_template>
+ *    <name>$QueryTemplateName</name>
+ *    <query><![CDATA[$SQLQuery]]></query>
+ *    <parameter_types>$ParameterType1,$ParameterType2,...</parameter_types>
+ *    <parameter_values>
+ *      <value>$ParameterValueA1, $ParameterValueA2,...</value>
+ *      <value>$ParameterValueB1, $ParameterValueB2,...</value>
+ *      ...
+ *    </parameter_values>
+ *  </query_template>
+ *  ...
+ * <query_templates>
+ * where $ParameterType is the integer java.sql.Types value (i.e., 4 for
+ * integer, 16 for boolean etc.) and each value tag within 'parameter_values'
+ * contains the values for one instantiation of the parameters set in $SQLQuery. 
+ * The SQL query string is read as a PreparedStatement, i.e., parameters are 
+ * defined in the string via a '?' placeholder. An example for a templated
+ * benchmark can be found in data/templated/example.xml. The file path for the 
+ * XML template has to be defined in the workload configuration using the
+ * 'query_templates_file' tag. An example configuration can be found in
+ * config/sqlserver/sample_template_config.xml. The example can be executed if a
+ * loaded TPC-C instance is used as JDBC endpoint.
+ * 
+ * Templated benchmarks are instantiated using 'templated' as benchmark class
+ * when running BenchBase via the command line.
+ */
```

Comment:
Maybe move this into a `README.md` in the directory along with some links to examples and just leave a comment here to read the README?